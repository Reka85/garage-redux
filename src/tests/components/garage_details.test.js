import React from "react";
import  { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import GarageDetails from "../../components/garage_details";

test("should render GarageDetails correctly", () => {
  const wrapper = shallow(<GarageDetails garage={"MyGarage"}/>);
  expect(wrapper).toMatchSnapshot();
});
