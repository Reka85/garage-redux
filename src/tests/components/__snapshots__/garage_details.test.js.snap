// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GarageDetails correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GarageDetails
    garage="MyGarage"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="garage-photo"
        />,
        <h2>
          MyGarage
        </h2>,
        <Link
          replace={false}
          to="/new"
        >
          <button>
            Add new cars
          </button>
        </Link>,
      ],
      "className": "garage-details",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "garage-photo",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "MyGarage",
        },
        "ref": null,
        "rendered": "MyGarage",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Add new cars
          </button>,
          "replace": false,
          "to": "/new",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add new cars",
          },
          "ref": null,
          "rendered": "Add new cars",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="garage-photo"
          />,
          <h2>
            MyGarage
          </h2>,
          <Link
            replace={false}
            to="/new"
          >
            <button>
              Add new cars
            </button>
          </Link>,
        ],
        "className": "garage-details",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "garage-photo",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MyGarage",
          },
          "ref": null,
          "rendered": "MyGarage",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Add new cars
            </button>,
            "replace": false,
            "to": "/new",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add new cars",
            },
            "ref": null,
            "rendered": "Add new cars",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
